### PR TITLE
chore: use logger in advanced example, make sure all instanceof conditionals are exhaustive

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

File: examples/advanced.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ const cacheValue = 'value';
 // you can customize your log level or provide your own logger factory to
 // integrate with your favorite logging framework
 const loggerFactory: MomentoLoggerFactory = new DefaultMomentoLoggerFactory();
+const logger = loggerFactory.getLogger(`AdvancedExample`);
```
Comment (malandis) [inline examples/advanced.ts:22]: GitHub complains about using backticks when not interpolating

```suggestion
const logger = loggerFactory.getLogger('AdvancedExample');
```

File: examples/advanced.ts [hunk 2]
```diff
@@ -47,67 +48,75 @@ async function createCacheExample() {
...
-      console.log(
+      logger.info(
         `Concurrent get for ${key} returned ${response.valueString()}`
       );
     } else {
-      console.log(
+      logger.info(
         `Something went wrong with concurrent get for key ${key}: ${response.toString()}`
       );
     }
```
Comment (malandis) [inline examples/advanced.ts:122]: Not a blocker but do we want to handle the error case separately from an unknown response?

Comment (cprice404) [reply examples/advanced.ts:122]: I kind of wanted to illustrate that the user can choose which level they want to handle it at by including some examples of both.  wdyt?

**Comment by cprice404** [comment]
rebased to fix merge conflict.

**Review by nand4011 — APPROVED** [review]

File: examples/advanced.ts (hunk 2 above)
Comment (malandis) [reply examples/advanced.ts:122]: Sounds good. Just wanted to make sure it wasn't an oversight

**Review by malandis — APPROVED** [review]

**Pull request merged**